src/NodeCompiler/Exception/UnableToCompileNode.php

```diff
@@ -5,7 +5,56 @@
 namespace Roave\BetterReflection\NodeCompiler\Exception;
 
 use LogicException;
+use PhpParser\Node;
+use Roave\BetterReflection\NodeCompiler\CompilerContext;
+use Roave\BetterReflection\Reflection\ReflectionClass;
+use function get_class;
+use function sprintf;
 
 class UnableToCompileNode extends LogicException
 {
+    public static function forUnRecognizedExpressionInContext(Node\Expr $expression, CompilerContext $context) : self
+    {
+        return new self(sprintf(
+            'Unable to compile expression in %s: unrecognized node type %s at line %d',
+            self::compilerContextToContextDescription($context),
+            get_class($expression),
+            $expression->getLine()
+        ));
+    }
+
+    public static function becauseOfNotFoundClassConstantReference(
+        CompilerContext $fetchContext,
+        ReflectionClass $targetClass,
+        Node\Expr\ClassConstFetch $constantFetch
+    ) : self {
+        /** @var Node\Identifier $constantFetch ->name */
+        return new self(sprintf(
+            'Could not locate constant %s::%s while trying to evaluate constant expression in %s at line %s',
+            $targetClass->getName(),
+            $constantFetch->name->name,
+            self::compilerContextToContextDescription($fetchContext),
+            $constantFetch->getLine()
+        ));
+    }
+
+    public static function becauseOfOfNotFoundConstantReference(
+        CompilerContext $fetchContext,
+        Node\Expr\ConstFetch $constantFetch
+    ) : self {
+        /** @var Node\Identifier $constantFetch ->name */
+        return new self(sprintf(
+            'Could not locate constant "%s" while evaluating expression in %s at line %s',
+            reset($constantFetch->name->parts),
+            self::compilerContextToContextDescription($fetchContext),
+            $constantFetch->getLine()
+        ));
+    }
+
+    private static function compilerContextToContextDescription(CompilerContext $fetchContext) : string
+    {
+        return $fetchContext->hasSelf()
+            ? $fetchContext->getSelf()->getName()
+            : 'unknown context'; // @TODO review feedback here, plox!
```

Comment:
This is probably a problem

Reply:
If the context is in a function?

Reply:
`CompilerContext` works for functions - basically, if `$fetchContext->hasSelf()` is `false`, it's a function AFAIK

Reply:
Yes, we're missing which function we're in (if any) but as you say that's only useful for exceptions. Maybe make a new issue to improve on this and add that capability to `CompilerContext`?

Reply:
I'll open an issue then. Can you meanwhile drop the `@TODO` blocks in this PR?